pages/api/notify.ts

```diff
@@ -0,0 +1,51 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+
+const projectId = process.env.NEXT_PUBLIC_PROJECT_ID;
+if (!projectId) {
+  throw new Error("You need to provide NEXT_PUBLIC_PROJECT_ID env variable");
+}
+
+const projectSecret = process.env.CAST_PROJECT_SECRET;
+if (!projectSecret) {
+  throw new Error("You need to provide CAST_PROJECT_SECRET env variable");
+}
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  if (req.method !== "POST") {
+    throw new ReferenceError("Method not allowed");
+  }
+
+  const notificationPayload = req.body;
+  if (!notificationPayload) {
+    return res.status(400).json({ success: false });
+  }
+
+  try {
+    // We can construct the URL to the Cast server using the `castUrl` property
+    // of the `PushDappClient` (which will be `https://cast.walletconnect.com` by default),
+    // together with our Project ID.
+    const result = await fetch(
+      `https://cast.walletconnect.com/${projectId}/notify`,
+      {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+          Authorization: projectSecret as string,
+        },
+        body: JSON.stringify(notificationPayload),
+      }
+    );
+
+    const gmRes = await result.json(); // { "sent": ["eip155:1:0xafeb..."], "failed": [], "not_found": [] }
+    const isSuccessfulGm = gmRes.sent.includes(notificationPayload.accounts[0]);
```

Comment:
Seems like we're using different criteria for success now, in the manual GM send it's `gmRes.success` but here it still seems to be the specific account check

Reply:
My bad I forgot to update the API route. What's happening now is:
- Send manual gm
- Request to Next API route /notify
- From internal  /notify we make a request to Cast with project secret and check if successful based on account
- Front-end checks if successful based on `success` res from /notify

Reply:
Ah makes sense, missed this [line diff](https://github.com/WalletConnect/gm-dapp/pull/13/files#diff-4f3679ee680d8f5a798f6ddff60b89dd58f79afd5ea62818f8f846efb38e6b21R160) hadn't thought through the flow given the additional guarded header 👍